src/arrayMethodSort.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const ARRAY = [...this];
+    let cmpFunc = compareFunction;
+
+    if (!compareFunction) {
+      cmpFunc = (a, b) => String(a) > String(b);
```

Comment:
The default comparator should return a number less than 0, 0, or greater than 0. Using '>' will only return a boolean which may not work correctly for all cases. Consider using `localeCompare` for string comparison, which returns a number according to the sort order.